easybuild/easyconfigs/r/RAxML-NG/RAxML-NG-0.9.0-foss-2019b.eb

```diff
@@ -0,0 +1,56 @@
+easyblock = 'CMakeMakeCp'
+
+name = 'RAxML-NG'
+version = '0.9.0'
+
+homepage = "https://github.com/amkozlov/raxml-ng"
+description = """RAxML-NG is a phylogenetic tree inference tool which
+uses maximum-likelihood (ML) optimality criterion. Its search heuristic
+ is based on iteratively performing a series of Subtree Pruning and Regrafting
+ (SPR) moves, which allows to quickly navigate to the best-known ML tree."""
+
+toolchain = {'name': 'foss', 'version': '2019b'}
+
+sources = [{
+    'filename': 'raxml-ng-%(version)s.tar.gz',
+    'git_config': {
+        'url': 'https://github.com/amkozlov',
```

Comment:
When downloading from GitHub, you don't need to use `git_config` at all (which is troublesome w.r.t. checksums, as @migueldiascosta pointed out).

I suggest to use this instead:

```python
source_urls = ['https://github.com/amkozlov/raxml-ng/archive/']
sources = ['%(version)s.tar.gz']
```

(and fix the checksum)

This corresponds to the exact same sources as when using tag `0.9.0` tag, it's just a static tarball provided by GitHub (static in the sense that it'll be the exact same download for everyone, which is not the case when using `git_config` where the tarball is created with the local `tar` and `gzip` commands).

If you care strongly about the source filename, you can make EasyBuild rename it after download:

```python
sources = ['download_filename': '%(version)s.tar.gz', 'filename': 'raxml-ng-%(version)s.tar.gz']
```

or

```python
sources = ['download_filename': '%(version)s.tar.gz', 'filename': SOURCELOWER_TAR_GZ]
```

Reply:
Hello @migueldiascosta , hello @boegel ,

Thanks both for your feedback. Concerning the tool chain and the list of dependencies, i will try to do that and update the pull request.

Concerning the checksum, we knew that could cause some problems, but we don't know how to proceed otherwise. The archive doesn't contain the full source code of the solution. Some libraries are also necessary. That's why we added the recursive True to the git_config.

We tried to specify all the sources in the list of sources and source_urls, but that doesn't work out. I don't know if there's a means of specifying manually where to decompress a source within the directory tree of another source.

Another alternative could be to create easy config files for all the additional libraries and then set them up as dependencies, but i don't think that makes sense, since they don't have enough specific weight to be compiled or deployed independently, and the final result of RAxML-NG compilation is a single executable statically linked.

Would you have any suggestions about this?

Best Regards

Reply:
You can list everything in `sources`, and the move things into the correct place before you start the build.

Something like this (untested and clearly needs work, but I hope you get the idea):

```python
sources = [
    {
        'source_urls': ['https://github.com/amkozlov/raxml-ng/archive/'],
        'filename': '%(version)s.tar.gz',
    },
    {
        'source_urls': ['https://github.com/amkozlov/terraphast-one/archive',
        'download_filename': '6dcb061.tar.gz',
        'filename': 'terraphast-one-20180618.tar.gz',
    },
]

# copy dependencies into place
preconfigopts = "cp -a %(builddirs)/terraphast-one %(builddir)s/raxml-ng/libs/terraphast && "
```

The directory names `terraphast-one` and `raxml-ng` are probably not fully correct, they may have a version/commit ID attached, but you may be able to work around that using `terraphast-one*` and `raxml-ng*`.

I hope this helps, but do let us know if you're stuck or have questions.